### ISSUE TITLE
Re-add channel normalization and pixel thresholding

### ISSUE THREAD

**Issue by alex-l-kong** [angelolab/ark-analysis#980]
**Is your feature request related to a problem? Please describe.**

Candace has created her own repo for the Pixie paper. It's been out for some time now, so now would be a good idea to add the channel and pixel normalization strategies back.

**Describe the solution you'd like**

#913 removed this, so just undo the changes.

**Additional context**

This will be helpful for testing Noah's TONIC cohort on the updated `ark-analysis` Docker image.

**Comment by cliu72** [comment]
@ngreenwald I was getting asked why we added a 99.9th normalization before the frequency normalization, and so to check I did a cluster consistency score comparison. y-axis here is cluster consistency score, and distribution is all pixels in my 12 FOV HIV test set. Smaller cluster consistency score with lower variance is better. Adding the 99.9 before seems to make it a little worse - why did you decide to add it again?

![image](https://user-images.githubusercontent.com/11382352/235269288-9ea84a61-16ad-4171-97c0-31c988875349.png)

**Comment by alex-l-kong** [comment]
Currently in this repo, there is no channel normalization or pixel thresholding since we removed it for your paper (#913). The original plan was to add them back after your paper was finalized, seems like we may want to hold off further for now?

I do need to re-run some data through `ark-analysis` (including channel norm and pixel thresholding), but I can create a local branch to do that instead.

**Comment by ngreenwald** [comment]
I found that super bright channels like SMA were dominating the clustering without a prior normalization. So it makes sense that the consistency would look better without this step. It's much easier to just assign SMA to all pixels that have even a bit of SMA in them. 

This allowed me to get out other phenotypes besides just SMA for pixels that were co-expressing, or nearby, SMA signal.

**Comment by cliu72** [comment]
@ngreenwald Ah yes, I remember now. I don't think that's the reason the cluster consistency score is higher for this HIV dataset though - there is nothing else in the panel that is co-expressed with SMA and there are very few SMA expressing pixels in general. I don't think that there is a similarly "dominating" marker in this panel - the only pixels with a lot of co-expression are the myeloid ones and none of those markers are particularly bright. Conceptually, I also don't totally understand - why is that problem not being taken care of by the 99.9 normalization after the pixel normalization?

Hadeesha is seeing a pretty significant time increase in the pre-processing step when these steps were included. I think he said it took it more than 65 hours for either the DCIS or GBM cohort (I can't remember which one), just for the pre-processing step before clustering happens. So I'm wondering if this step is really that necessary.  I'm wondering if the effect you were seeing was just a random seed thing? Did you test this on any other datasets?

**Comment by ngreenwald** [comment]
Nope, I didn't test it on any other datasets. But the both of the dominating markers (SMA and Vimentin) had this problem initially, and the pre-normalization fixed it. 

If you find that it was something specific to my dataset, totally fine to not add it back. However, calculating these percentiles shouldn't take that long since it uses a subset of the FOVs, and normalizing each image is extremely fast once it's already loaded into memory, so my suspicion is that something else is causing the slowdown?

**Comment by cliu72** [comment]
I think the latest version of the code before we took it out for my paper was using every FOV for the percentile calculation (not a subset): https://github.com/angelolab/ark-analysis/blob/v0.5.2/src/ark/phenotyping/pixel_cluster_utils.py

It's basically reading in every channel in every fov to first calculate the normalization percentiles, then reading in every channel in every fov again to calculate the low expressing threshold, and then reading in every channel in every fov one more time to do the actual gaussian blurring/normalization. I think this is causing the slowdown.

I'll do some more tests with different datasets to see if I find that dominating markers are causing an issue. I don't think it's a your dataset issue since a lot of people have SMA and Vimentin - I'm wondering if maybe there's something that we can do normalize super bright markers that doesn't involve reading in every single image.

**Comment by cliu72** [comment]
@alex-l-kong Talked to Noah about this, and for now, we're going to add back in those two normalization steps, but change both of them so that they only use a subset of the total FOVs (similar to what you did earlier for `compute_pixel_cluster_channel_avg`. I think default number of FOVs can be 100, same as what you did for  `compute_pixel_cluster_channel_avg`.
